XPCBackupService/Services/BackupRealm.swift

```diff
@@ -8,14 +8,20 @@
 import Foundation
 import RealmSwift
 
-protocol BackupRealmProtocol {
-    func getRealm() throws -> Realm?
-    func addSyncedNode(_ node: SyncedNode) throws
+//protocol BackupRealmProtocol {
+//    func getRealm() throws -> Realm?
+//    func addSyncedNode(_ node: SyncedNode) throws
+//    func findSyncedNode(url: URL, deviceId: Int) -> ThreadSafeReference<SyncedNode>?
+//    func editSyncedNodeDate(remoteUuid: String, date: Date) throws
+//}
+
+protocol BackupRealmProtocol: GenericRepositoryProtocol where T == SyncedNode {
```

Comment:
you don't need to specify "subprotocols", instead, you can work by creating a class that implements a `GenericRepositoryProtocol`, and add to the class the members that are not related to the base protocol itself.

The protocol should only have members that are abstract enough to be used in whatever repository